### PR TITLE
[fix](map) fix core dump when upgrading from 1.2.x to 2.0.x with map datatype column 

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 1]
```diff
@@ -137,6 +137,7 @@ Status ColumnReader::create(const ColumnReaderOptions& opts, const ColumnMetaPB&
         }
         case FieldType::OLAP_FIELD_TYPE_MAP: {
             // map reader now has 3 sub readers for key, value, offsets(scalar), null(scala)
+            DCHECK(meta.children_columns_size() == 3 || meta.children_columns_size() == 4);
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:140]: why not only 3?

File: be/src/olap/rowset/segment_v2/column_writer.cpp [hunk 2]
```diff
@@ -262,6 +262,11 @@ Status ColumnWriter::create(const ColumnWriterOptions& opts, const TabletColumn*
         }
         case FieldType::OLAP_FIELD_TYPE_MAP: {
             DCHECK(column->get_subtype_count() == 2);
+            if (column->get_subtype_count() < 2) {
+                return Status::InvalidArgument(
+                        "If you upgraded from version 1.2.*, please drop the "
+                        "map column and wait a while to re-create the map column.");
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/column_writer.cpp:268]: "drop the MAP column and wait a while to re-create the map column" => "DROP the MAP columns and then ADD the MAP columns back."

File: be/src/olap/rowset/segment_v2/column_writer.cpp [hunk 3]
```diff
@@ -262,6 +262,11 @@ Status ColumnWriter::create(const ColumnWriterOptions& opts, const TabletColumn*
         }
         case FieldType::OLAP_FIELD_TYPE_MAP: {
             DCHECK(column->get_subtype_count() == 2);
+            if (column->get_subtype_count() < 2) {
+                return Status::InvalidArgument(
+                        "If you upgraded from version 1.2.*, please drop the "
```
Comment (eldenmoon) [inline be/src/olap/rowset/segment_v2/column_writer.cpp:267]: return Status::InternalError to print stack, InvalidArgument will not print stack

File: be/src/olap/rowset/segment_v2/column_reader.cpp (hunk 1 above)
Comment (amorynan) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:140]: if map column is nullable, meta.children_columns(3) means to read null map

**Comment by amorynan** [comment]
run buildall

**Pull request merged**